src/windows/main_window.py

```diff
@@ -2294,12 +2294,21 @@ def load_recent_menu(self):
             self.recent_menu.clear()
 
         # Add recent projects to menu
-        for file_path in reversed(recent_projects):
-            new_action = self.recent_menu.addAction(file_path)
-            new_action.triggered.connect(functools.partial(self.recent_project_clicked, file_path))
+        if recent_projects:
+            for file_path in reversed(recent_projects):
+                # Add each recent project
+                new_action = self.recent_menu.addAction(file_path)
+                new_action.triggered.connect(functools.partial(self.recent_project_clicked, file_path))
+
+            # Add 'Clear Recent Projects' menu to bottom of list
+            self.recent_menu.addSeparator()
+            self.recent_menu.addAction(self.actionClearRecents)
+            self.actionClearRecents.triggered.connect(self.clear_recents_clicked)
+        else:
+            self.recent_menu.addAction(_("No Recent Projects"))
```

Comment:
Just in the interest of keeping the indentation down...
```suggestion
        # Show just a placeholder menu, if we have no recent projects list
        if not recent_projects:
            self.recent_menu.addAction(_("No Recent Projects"))
            return

        for file_path in reversed(recent_projects):
            # Add each recent project
            new_action = self.recent_menu.addAction(file_path)
            new_action.triggered.connect(functools.partial(self.recent_project_clicked, file_path))

        # Add 'Clear Recent Projects' menu to bottom of list
        self.recent_menu.addSeparator()
        self.recent_menu.addAction(self.actionClearRecents)
        self.actionClearRecents.triggered.connect(self.clear_recents_clicked)
```

Reply:
OK, I guess maybe it's only 9 lines, _technically_. :roll_eyes: